File: cupstonefrontend/gameshop/src/app/models/carrello.ts
import { Prodotto } from "./prodotto";
import { Utente } from "./utente";

export interface Carrello {
    id: number;
    prodotti: Prodotto[]; // Lista di identificatori di prodotti
    totale: number;
    utente: Utente; // Identificatore dell'utente
}

File: capstone/src/main/java/com/capstone/capstone/entity/Carrello.java
package com.capstone.capstone.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
@Table(name="carrelli")
public class Carrello {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToMany
    @JoinTable(
            name="prodotti_carrello",
            joinColumns = @JoinColumn(name="carrello_id"),
            inverseJoinColumns = @JoinColumn(name="prodotto_id")

    )
    private List<Prodotto> prodotti;
    private double totale;
    @ManyToOne
    @JoinColumn(name="utente_id")
    private Utente utente;
}